### ISSUE TITLE
How to use Local Storage for backup

### ISSUE THREAD

**Issue by Schnuecks** [Requarks/wiki#1288]
Hello,

how could i use the local storage for backup?
i ran the docker-compose.yml from docs.

how do i need to modify the docker-compose.yml to set a volume or path so wikijs could backup in it and copy to the host.

OS: CentOS 7, docker-compose
Wiki.js version:  2.0.12
Database engine: postgres 9

**Comment by NGPixel** [comment]
You would need to mount a folder, e.g.:

```yaml
version: "3"
services:

  db:
    image: postgres:11-alpine
    environment:
      POSTGRES_DB: wiki
      POSTGRES_PASSWORD: wikijsrocks
      POSTGRES_USER: wikijs
    logging:
      driver: "none"
    restart: unless-stopped
    volumes:
      - db-data:/var/lib/postgresql/data

  wiki:
    image: requarks/wiki:2
    depends_on:
      - db
    environment:
      DB_TYPE: postgres
      DB_HOST: db
      DB_PORT: 5432
      DB_USER: wikijs
      DB_PASS: wikijsrocks
      DB_NAME: wiki
    restart: unless-stopped
    ports:
      - "80:3000"
    volumes:
      - /host/backup/path:/backups

volumes:
  db-data:
```

Then specify `/backups` at the path in Wiki.js admin

**Comment by Schnuecks** [comment]
Hello,

i tried that and save it in the compose file. the folder on host and container is created.

i specified the path in wikijs but nothing is happen if i click on create backup.

both folder are empty.

**Comment by ChristopherRotter** [comment]
I ran into the same problem.

This

`wiki:
    volumes:
      - ./backup:/backups`

did the trick for me.

The action _Create backup_ works for me, however, **_Import Everything_ does not restore any pages I delete...**

**Comment by Schnuecks** [comment]
That still doesn't work for me.The backups folder defined in docker-compose.yml is created as root.
So the user node who runs doesn't have the permission to write to that folder.
If i backup to the node home directory it works.

**Comment by ssimontis** [comment]
@Schnuecks Assign the user node to the Docker group and that should fix your issue.

**Comment by Schnuecks** [comment]
@ssimontis How? I have no node user on the host who could i assigne for docker group

**Comment by ssimontis** [comment]
@Schnuecks I'll get you an answer tonight, one of my goals for this evening is to dig into the Dockerfiles

**Comment by krema** [comment]
The problem is that the backup folder is created with root ownership on the container and therefore the node user does not have write permissions on this folder. To fix this the folder ownership in the container must be changed to node:node. We could change the Dockerfile to have a default backup folder with the correct permissions in the container. What do you think?

**Comment by ssimontis** [comment]
I think that sounds like a winner to me...in my opinion the more directories we can pry away from root ownership related to our applications, the better

**Comment by Schnuecks** [comment]
Sounds good. A default backup location with correct permission seems the way to go.

**Comment by NGPixel** [comment]
`/wiki/data/content` has been added as a predefined volume with `node:node` permissions in the Dockerfile. It will be available in the 2.1 release.
